lib/SIL/LinearLifetimeChecker.cpp

```diff
@@ -499,8 +499,6 @@ LinearLifetimeError LinearLifetimeChecker::checkValue(
     SILValue value, ArrayRef<Operand *> consumingUses,
     ArrayRef<Operand *> nonConsumingUses, ErrorBehaviorKind errorBehavior,
     SmallVectorImpl<SILBasicBlock *> *leakingBlocks) {
-  assert(!consumingUses.empty() && "Must have at least one consuming user?!");
```

Comment:
I think this assert is an important sanity check for understanding how the verifier works.... Do you need to add another case?
`assert(!consumingUses.empty() || nonConsumingUses.empty()`

Reply:
The case I saw in 'Plank' has empty consuming uses and non-empty nonConsuming uses.
It's an alloc_stack which gets used in an apply and there's an unreachable at the end of the block.

Reply:
The destroy_addr were eliminated earlier by NoReturnFolding

Reply:
I see, then wouldn't this make sense? `!consumingUses.empty() || !deadEndBlocks.empty()`

Reply:
Yes, that's the right thing to do.